### ISSUE TITLE
The state of the “Turn on/off Night Mode” feature is not synced with the manual theme picker selection from Theme Settings

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-mobile/firefox-ios#19094]
**[Affected versions]:**

* Firefox Release 124.0 (39587).

**[Affected Platforms]:**

* iOS 16.4 - iPhone 12 Pro.

**[Preconditions]:**

* System Light theme is activated (from device Settings > Display & Brightness).

**[Steps to reproduce]:**

1. Open the Firefox app;
1. From the Hamburger menu, select “Turn on Night Mode”;
1. From the Hamburger menu, select Settings > Theme and toggle the “Use System Light/Dark Mode” option off;
1. Observe the behavior.

**[Expected result]:**

* The “Dark” theme is preselected from the “Theme Picker” section.

**[Actual result]:**

* The “Light” theme is preselected from the “Theme Picker” section.

**[Notes]:**

* This issue could not be reproduced with Firefox Release 123.0 (38801);

* Because of this issue, the theme can’t be manually changed in this situation (from Settings > Theme) until the user turns off Night Mode first.

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/FXIOS-8603)

**Comment by data-sync-user** [comment]
➤ Norberto Andres Furlan commented:

Roux Buciu can you check this one?

**Comment by data-sync-user** [comment]
➤ Roux Buciu commented:

👀

**Comment by dicarobinho** [comment]
I am able to reproduce this issue. Looking on it...

**Comment by data-sync-user** [comment]
➤ Alexandru Farcasanu commented:

Roux Buciu I need some input here…

What should be the normal behaviour here?

Light/Dark options from “Settings > Theme” should have exactly the same behaviour as Turn ON/OFF Night Mode from Hamburger Menu?

I mean, should change the entire theme of the app, as Turn ON/OFF Night Mode it do?

**Comment by data-sync-user** [comment]
➤ Roux Buciu commented:

So, technically, this is a finicky bit with Night Mode.

“Settings > Theme” will change app chrome to light/dark.

Night Mode will change webkit content to dark and app chrome to dark (regardless of theme settings option). No matter what you set in Settings, if Night Mode is on, your app will be dark.

Now, technically, the theme picker should say what the user’s last selection was - and it does. But how does Night Mode affect that? And, also, given that you’re in Night Mode, none of your theme settings will apply, so no matter what you do, the app is in Night Mode. So to really fix how all this works, we should figure out what UX we want. We’re looking at a few things with Night Mode, so, before worrying about this edge case, we’re gonna have to get some answers from the investigation carried out.

In other words, let’s not worry about this for now (especially as it’s an edge case), until we have further answers from Norberto Andres Furlan re the Night Mode investigation.

**Comment by data-sync-user** [comment]
➤ Norberto Andres Furlan commented:

I am closing this ticket for now. I already launch the experiment. It is a very small edge case that I don’t think it will change the experiment. Thanks anyway!
